### DISCUSSION TITLE
Diffraction Pressure Distribution

### DISCUSSION THREAD

**Discussion by ckesanapalli** [capytaine/capytaine#93] [Q&A] [answered]
Can you tell me, how to output Diffraction Pressure Distribution on the body surface in capytaine?

**Comment by mancellin** [comment, marked as answer]
For linear potential flow, the pressure is proportional to the potential.
You can access the potential on each panel of the floating body in the `DiffractionResult` object, so something like
```python
problem = DiffractionProblem(...)
result = cpt.BEMSolver(...).solve(problem, keep_details=True)
result.potential
```

**Reply by ckesanapalli** [reply]
Can you tell me how to plot a 3D surface with a pressure color map? I am trying to do it using `all_bodies.show_matplotlib(facecolors=plt.cm.jet(np.absolute(result.potential)))` but it is showing only single color. The only option that I can do is by using `scatter3D` and plot the face centers. But would really like the surface plots similar to TECPLOT pressure plots.

**Reply by ckesanapalli** [reply]
I resolved it. it is due to the potential maxima and minima. If the maxima and minima values are large and the difference is small then it doesn't show the pressure map. I copied the `all_bodies.show_matplotlib` source code and modified it by normalizing the potential to 0, 1, and then plotted it. If possible, I will pull request some suggestions regarding the plot.
